src/bundle/Controller/LocationController.php

```diff
@@ -144,12 +144,16 @@ private function deleteLocationsBasedOnFormSubmit(FormInterface $form, $redirect
         return false;
     }
 
-    private function addLocationBasedOnFormSubmit(FormInterface $form, Content $content)
+    private function addLocationsBasedOnFormSubmit(FormInterface $form, Content $content)
     {
         if ($form->get('add')->isClicked()) {
-            $this->uiLocationService->addLocation(
-                $content->contentInfo,
-                $form->get('parentLocationId')->getData()
+            $locationService = $this->uiLocationService;
```

Comment:
You don't need to do that, you can use `$this->uiLocationService` from the callback.